File: src/app/models/loans.ts
export class Loans {
  loanId!: number;
  pinNo!: string;
  bookId!: number;
  issueDate!: string;  // ISO string date
  dueDate!: string;    // ISO string date
  returnDate!: string | null;
  status!: string; // 'ISSUED', 'OVERDUE', 'RETURNED'
}

File: library-manager/src/main/java/com/example/demo/models/loans.java
package com.example.demo.models;

import java.time.LocalDate;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class loans {
	
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int loanId;
    
    @Column(name = "pin_no")
    private String pinNo;
    private int bookId;

    private LocalDate issueDate;
    private LocalDate dueDate;
    private LocalDate returnDate;

    private String status;


}